### PR TITLE
Add methods: groupBy, chunk, splice, find, findLast, any, every

### PR DESCRIPTION
**Pull request by dazet**

### CODE REVIEW

**Review by bronek89 — APPROVED** [review]

File: src/ArrayValue.php [hunk 1]
```diff
@@ -128,6 +139,11 @@ public function join(ArrayValue $other);
      */
     public function slice(int $offset, int $length);
 
+    /**
+     * @return ArrayValue
+     */
+    public function splice(int $offset, int $length, array $replacement = []);
```
Comment (bronek89) [inline src/ArrayValue.php]: `$replacement` should not be `ArrayValue` instead of simple `array` in this case?

**Comment by bronek89** [comment]
I'll add `chunk`, `splice`, `find`, `any` and maybe `groupBy` to `IterableValue` after merge of this PR and IterableValue

File: src/ArrayValue.php (hunk 1 above)
Comment (dazet) [reply src/ArrayValue.php]: I guess it will be more consistent :+1:

File: src/AssocArray.php [hunk 2]
```diff
@@ -195,11 +195,39 @@ public function last()
         return $this->values()->last();
     }
 
+    /**
+     * @param callable $filter function(mixed $value): bool { ... }
```
Comment (dlakomski) [inline src/AssocArray.php]: Above you removed these `{ ... }` blocks, but here it remains.

File: src/PlainArray.php [hunk 3]
```diff
@@ -227,11 +255,63 @@ public function last()
         return $count > 0 ? $this->items[$count - 1] : null;
     }
 
+    /**
+     * @param callable $filter function(mixed $value): bool { ... }
```
Comment (dlakomski) [inline src/PlainArray.php]: Same here.

**Review by dlakomski — APPROVED** [review]

**Comment by bronek89** [comment]
@dazet Will you merge this on today's midnight? :)

**Pull request merged**